File: Seminar/src/main/java/org/sopt/Seminar/domain/user/service/UserService.java
package org.sopt.Seminar.domain.user.service;

import lombok.RequiredArgsConstructor;
import org.sopt.Seminar.domain.user.dto.request.UserRequest;
import org.sopt.Seminar.domain.user.model.User;
import org.sopt.Seminar.domain.user.repository.UserRepository;
import org.springframework.security.crypto.password.PasswordEncoder;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@Service
@RequiredArgsConstructor
@Transactional(readOnly = true)
public class UserService {

    private final UserRepository userRepository;
    private final PasswordEncoder passwordEncoder;

    @Transactional
    public String create(UserRequest request) {
        User user = User.builder()
                .nickname(request.nickname())
                .password(passwordEncoder.encode(request.password()))
                .build();
        userRepository.save(user);

        return user.getId().toString();
    }

    public void signIn(UserRequest request) {
        User user = userRepository.findByNickname(request.nickname())
                .orElseThrow(() -> new RuntimeException("해당하는 회원이 존재하지 않습니다."));

        if (!passwordEncoder.matches(request.password(), user.getPassword())) {
            throw new RuntimeException("비밀번호가 일치하지 않습니다.");
        }
    }
}

File: kopring/week2/kotlin-semina/src/main/kotlin/org/sopt/kotlinsemina/domain/user/service/UserService.kt
package org.sopt.kotlinsemina.domain.user.service

import org.sopt.kotlinsemina.domain.member.exception.MemberNotFoundException
import org.sopt.kotlinsemina.domain.user.dto.request.UserRequest
import org.sopt.kotlinsemina.domain.user.model.User
import org.sopt.kotlinsemina.domain.user.repository.UserRepository
import org.springframework.data.repository.findByIdOrNull
import org.springframework.security.crypto.password.PasswordEncoder
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
@Transactional(readOnly = true)
class UserService(
    private val userRepository: UserRepository,
    private val passwordEncoder: PasswordEncoder
) {

    @Transactional
    fun create(request: UserRequest): String {
        val user = userRepository.save(
            User(
                nickname = request.nickname,
                password = request.password
            )
        )

        return user.id.toString()
    }

    fun signIn(request: UserRequest) {
        val user = userRepository.findByNickname(request.nickname)
            ?: throw RuntimeException("해당하는 회원이 없습니다.")
        if (!passwordEncoder.matches(request.password, user.password)) {
            throw RuntimeException("비밀번호가 일치하지 않습니다.")
        }
    }
}